dnsmessage.proto

```diff
@@ -41,45 +41,47 @@ message PBDNSMessage {
+  optional bytes from = 6;                      // DNS requestor (client) as 4 (IPv4) or 16 (IPv6) raw bytes in network byte order
+  optional bytes to = 7;                        // DNS responder (server) as 4 (IPv4) or 16 (IPv6) raw bytes in network byte order
   optional uint64 inBytes = 8;                  // Size of the query or response on the wire
   optional uint32 timeSec = 9;                  // Time of message reception (seconds since epoch)
   optional uint32 timeUsec = 10;                // Time of message reception (additional micro-seconds)
   optional uint32 id = 11;                      // ID of the query/response as found in the DNS header
 
   message DNSQuestion {
-    optional string qName = 1;
-    optional uint32 qType = 2;
-    optional uint32 qClass = 3;
+    optional string qName = 1;                  // Fully qualified DNS name (with trailing dot)
+    optional uint32 qType = 2;                  // https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+    optional uint32 qClass = 3;                 // Typically 1 (IN), see https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-2
   }
-  optional DNSQuestion question = 12;
+  optional DNSQuestion question = 12;           // DNS query received from client
 
   message DNSResponse {
+    // See exportTypes in https://docs.powerdns.com/recursor/lua-config/protobuf.html#protobufServer
+    // for the list of supported resource record types.
     message DNSRR {
-      optional string name = 1;
-      optional uint32 type = 2;
-      optional uint32 class = 3;
-      optional uint32 ttl = 4;
-      optional bytes rdata = 5;
+      optional string name = 1;                 // Fully qualified DNS name (with trailing dot)
+      optional uint32 type = 2;                 // https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+      optional uint32 class = 3;                // Typically 1 (IN), see https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-2
+      optional uint32 ttl = 4;                  // TTL in seconds
+      optional bytes rdata = 5;                 // raw address bytes in network byte order for A & AAAA; text representation for others, with fully qualified (trailing dot) domain names
       optional bool  udr = 6;                   // True if this is the first time this RR has been seen for this question
     }
     optional uint32 rcode = 1;                  // DNS Response code, or 65536 for a network error including a timeout
-    repeated DNSRR rrs = 2;
+    repeated DNSRR rrs = 2;                     // DNS resource records in response
     optional string appliedPolicy = 3;          // Filtering policy (RPZ or Lua) applied
-    repeated string tags = 4;                   // Additional tags
+    repeated string tags = 4;                   // Additional tags applied
```

Comment:
I'm not sure whether "applied" makes sense for the tags. In dnsdist these tags are not tied to any policy or filtering.